docs/source/kedro_project_setup/dependencies.md

```diff
@@ -59,7 +36,36 @@ This installs Kedro and dependencies related to the data type group. An example
 To limit installation to dependencies specific to a data type:
 
 ```bash
-pip install "kedro-datasets[<group>.<dataset>]"
+pip install "kedro-datasets[<group>-<dataset>]"
+```
+
+For example, your workflow might require use of the `pandas.ExcelDataset`, so to install its dependencies, run `pip install "kedro-datasets[pandas-exceldataset]"`.
+
+```{note}
+From `kedro-datasets` version 3.0.0 onwards, the names of the optional dataset-level dependencies have been normalised to follow [PEP 685](https://peps.python.org/pep-0685/). The '.' character has been replaced with a '-' character and the names are in lowercase. For example, if you had `kedro-datasets[pandas.ExcelDataset]` in your requirements file, it would have to be changed to `kedro-datasets[pandas-exceldataset]`.
+```
+
+
+## Reproducible environments
+In order to make sure that the project dependencies and the transitive dependencies are pinned to specific versions, you can use [`pip-tools`](https://pypi.org/project/pip-tools/) to compile the `requirements.txt` file into a `requirements.lock` file.
+To install `pip-tools` in your virtual environment, run the following command:
```

Comment:
I wonder if we should mention that `uv pip compile` will replace this workflow and become the dominant way to do this...

https://github.com/astral-sh/uv

Reply:
As much as I love `uv`, it's a bit of a moving target ATM and it's not clear if they're even retaining the `uv pip` namespace https://github.com/astral-sh/uv/issues/1657

To give an idea of how young it is, nobody has yet asked for a 1.0 version yet 😂 (classic open source behavior)

I'd wait a bit to see where the project goes.

Reply:
Also there are efforts to standardize lock files, but the person writing the PEP became a dad :) https://fosstodon.org/@brettcannon/112203862546870204